src/pages/ModuleEditor/ModuleEditor.tsx

```diff
@@ -37,12 +38,16 @@ export default function ModuleEditor({match: {params: {moduleId}}}: Props) {
   const [message, setMessage] = useMessage();
   const [editing, setEditing] = React.useState(false);
   const [redirect, setRedirect] = React.useState();
+  const [canDisable, setCanDisable] = React.useState(false);
 
   function completeLoading() {
     setLoading(false);
     setMessage(undefined);
   }
 
+  const userRole = useSelector((state: any) => state.entities.currentUser.role);
+  const userId = useSelector((state: any) => state.entities.currentUser.id);
```

Comment:
This useSelector isn't reusable, let's change that.
create a new file `src/redux/selectors/hooks.ts`, 

add `useUser` function: since you are using multiple properties of the user, you might as well use the whole object.
```
export function useUser(): User | null {
    return useSelector((state: WebState) => getCurrentUser(state));
}
```
In this file use it:
```
const user = useUser();
```

Rewrite your usages of `userId` for `user?.id`,
and `role` as `user?.role`.